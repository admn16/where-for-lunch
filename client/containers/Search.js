import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Button from 'components/Button/Button';
import Condition from 'components/Condition/Condition';
import conditionActions from 'actions/conditionActions';
import placeActions from 'actions/placeActions';

export class Search extends PureComponent {
  static propTypes = {
    condition: PropTypes.object,
    place: PropTypes.object,
    setRadius: PropTypes.func.isRequired,
    fetchPlaces: PropTypes.func.isRequired,
  };

  handleOnConditionChange = (value) => {
    this.props.setRadius(value);
  }

  handleOnFormSubmit = (e) => {
    e.preventDefault();

    this.props.fetchPlaces(this.props.condition);
  };

  render() {
    const { condition } = this.props;
    const disableBtn = !(condition.latitude && condition.longitude);

    return (
      <form className="searchWrapper" onSubmit={this.handleOnFormSubmit}>
        <Condition condition={condition} action={this.handleOnConditionChange}/>
        <Button theme="homepageClick" disabled={disableBtn} />
      </form>
    );
  }
}

const mapStateToProps = state => ({
  condition: state.condition,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    setRadius: conditionActions.setRadius,
    fetchPlaces: placeActions.fetchPlaces,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Search);
