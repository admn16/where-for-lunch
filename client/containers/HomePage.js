import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Place from 'components/Place/Place';
import Search from 'containers/Search';

export class HomePage extends PureComponent {
  static propTypes = {
    place: PropTypes.object,
  };

  render() {
    const { place } = this.props;

    return (
      <div className="homePageWrapper">
        <Place place={place} />
        <Search />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  place: state.place,
});

export default connect(
  mapStateToProps,
)(HomePage);
