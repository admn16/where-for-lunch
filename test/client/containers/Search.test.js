import React from 'react';
import renderer from 'react-test-renderer';
import { Search } from 'containers/Search';

describe('Search Container', () => {
  const mockFn = jest.fn();
  const defaultProps = {
    condition: {},
    setRadius: mockFn,
    fetchPlaces: mockFn,
  };
  const searchComponent = props =>
    <Search {...{ ...defaultProps, ...props }}/>;

  test('renders properly', () => {
    const wrapper = renderer
      .create(searchComponent())
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
