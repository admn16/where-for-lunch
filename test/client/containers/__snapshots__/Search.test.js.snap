// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Container renders properly 1`] = `
<form
  className="searchWrapper"
  onSubmit={[Function]}
>
  <div
    className="root"
  >
    <span>
      Radius:
    </span>
    <div
      className="root"
    >
      <input
        onBlur={[Function]}
        type="text"
      />
    </div>
    <span>
      meters
    </span>
  </div>
  <button
    className="root homepageClick"
  />
</form>
`;
